src/mame/sega/suprloco.cpp

```diff
@@ -112,64 +148,93 @@ class suprloco_state : public driver_device
+  bit0 -> 1
+  bit1 -> 2
+  bit1 -> 3
+  bit2 -> 4
+  bit2 -> 5
+  bit2 -> 6
+  bit2 -> 7
+  GND -> 8
+
+  The other ends of all the resistors are tied together.
+  These 3 arrays then each connect to a Murata DST310-55Y5S222M100 EMI/RFI
+   Suppression Filter Disc (2200pF w/Ferrite Beads, 100V tolerance) and then
+   connect to the harness R, G and B contacts.
+  These same EMI/RFI suppression filters are used on all the input contacts
+   as well.
 
 ***************************************************************************/
 void suprloco_state::palette(palette_device &palette) const
 {
-	const uint8_t *color_prom = memregion("proms")->base();
+	const uint8_t *color_prom = memregion("colorprom")->base();
 
-	for (int i = 0; i < 512; i++)
+	for (int i = 0; i < 1024; i++)
 	{
 		int bit0, bit1, bit2;
 
 		// red component
 		bit0 = BIT(color_prom[i], 0);
 		bit1 = BIT(color_prom[i], 1);
 		bit2 = BIT(color_prom[i], 2);
-		int const r = 0x21 * bit0 + 0x47 * bit1 + 0x97 * bit2;
+		int const r = 0x24 * bit0 + 0x49 * bit1 + 0x92 * bit2;
 		// green component
 		bit0 = BIT(color_prom[i], 3);
 		bit1 = BIT(color_prom[i], 4);
 		bit2 = BIT(color_prom[i], 5);
-		int const g = 0x21 * bit0 + 0x47 * bit1 + 0x97 * bit2;
+		int const g = 0x24 * bit0 + 0x49 * bit1 + 0x92 * bit2;
 		// blue component
-		bit0 = 0;
+		bit0 = 0; // the low resistor in the blue array is simply left floating
 		bit1 = BIT(color_prom[i], 6);
 		bit2 = BIT(color_prom[i], 7);
-		int const b = 0x21 * bit0 + 0x47 * bit1 + 0x97 * bit2;
+		int const b = 0x24 * bit0 + 0x49 * bit1 + 0x92 * bit2;
 
 		palette.set_pen_color(i, rgb_t(r, g, b));
-
-		// hack: generate a second bank of sprite palette with red changed to purple
-		if (i >= 256)
-			palette.set_pen_color(i + 256, rgb_t(r, g, ((i & 0x0f) == 0x09) ? 0xff : b));
 	}
 }
 
 
-
 /***************************************************************************
 
   Callbacks for the TileMap code
 
 ***************************************************************************/
 
+/* tile attribute bits:
+
+	////////----------- low 8 bits of tile number
+	||||||||  /-------- unused
+	||||||||  |/------- connected to palette latch, but unused.
+	||||||||  ||/------ /FORCE_TILE_PRIORITY (if 0, this tile always appears above sprites)
+	||||||||  |||///--- palette selected for tile (note that bit 10(0xa) is used for BOTH the tile number AND the palette!)
+	||||||||  |||||///- high 3 bits of tile number
+	76543210  fedcba98
+*/
+
 TILE_GET_INFO_MEMBER(suprloco_state::get_tile_info)
 {
-	uint8_t const attr = m_videoram[2 * tile_index + 1];
+	uint16_t const tile_attr = (m_videoram[2*tile_index] | (m_videoram[2*tile_index+1]<<8));
 	tileinfo.set(0,
-			m_videoram[2 * tile_index] | ((attr & 0x03) << 8),
-			(attr & 0x1c) >> 2,
+			BIT(tile_attr,0,10),
+			((BIT(tile_attr,10,3) << 4) | 0x80 | (BIT(m_control,5,2) << 8)) >> 4, // this overlaps bit 10 intentionally
 			0);
-	tileinfo.category = (attr & 0x20) >> 5;
+	tileinfo.category = BIT(tile_attr,13);
```

Comment:
Please follow “English style” spacing rules – spaces after commas, spaces on both sides of binary and ternary operators, no space after right-associative unary operators, no space before left-associative unary operators.  Same for the rest of the file.

Reply:
this should be fixed now.